src/main/java/picard/sam/BamErrorMetric/BaseErrorCalculation.java

```diff
@@ -0,0 +1,266 @@
+         **/
+        @Override
+        public String getSuffix() {
+            return "error";
+        }
+
+        /**
+         * Returns the metric generated by the observed loci
+         **/
+        @Override
+        public ErrorMetrics.SimpleErrorMetric getMetric() {
+            return new ErrorMetrics.SimpleErrorMetric("", totalBases, nMismatchingBases);
+        }
+    }
+
+    /**
+     * A calculator that estimates the error rate of the bases it observes, assuming that the reference is truth.
+     * This calculator only includes bases that have been read twice in the same template (once from each read) and
+     * thus only includes bases that arise from the overlapping part of the reads. Over those bases the Calculator
+     * distinguishes between whether the two reads agree with each other but differ from the reference (indicative
+     * of a difference between the template and the reference, and when one of the reads agrees with the reference
+     * but the other does not which indicates that there might have been a sequencing error in that read.
+     */
+    public static class OverlappingReadsErrorCalculator extends BaseErrorCalculator {
+        long nBothDisagreeWithReference = 0;
+        long nDisagreeWithRefAndMate = 0;
+        long nThreeWaysDisagreement = 0;
+        long nTotalBasesWithOverlappingReads = 0;
+
+        static int currentPosition = 0;
+        static final Map<String, Set<RecordAndOffset>> readNameSets = new CollectionUtil.DefaultingMap<>(s -> new HashSet<>(), true);
+
+        private static void updateReadNameSet(final LocusInfo locusInfo) {
+            if (locusInfo.getPosition() == currentPosition) {
+                return;
+            }
+            readNameSets.clear();
+            locusInfo.getRecordAndOffsets().forEach(r -> readNameSets.get(r.getReadName()).add(r));
+            currentPosition = locusInfo.getPosition();
+        }
+
+        /**
+         * The function by which new loci are "shown" to the calculator
+         **/
+        @Override
+        public void addBase(final RecordAndOffset recordAndOffset, final SAMLocusAndReference locusAndRef) {
+            super.addBase(recordAndOffset, locusAndRef);
+            final byte readBase = recordAndOffset.getReadBase();
+            final SAMRecord record = recordAndOffset.getRecord();
+
+            // by traversing the reads and splitting into sets with the same name we convert a O(N^2) iteration
+            // into a O(N) iteration
+            updateReadNameSet(locusAndRef.getLocus());
+
+            final RecordAndOffset mate = readNameSets.get(record.getReadName())
+                    .stream()
+                    .filter(putative -> areReadsMates(record, putative.getRecord()))
+                    .findFirst()
+                    .orElse(null);
+
+            // we are only interested in bases for which the mate read also has a base over the same locus
+            if (mate == null) return;
```

Comment:
all `if`s should have `{` `}`s

Reply:
hmmm. really? I'm not sure that for such simple if statements this style has been adopted and agreed upon.

Reply:
Well, let's say _IMO_ they should. This is what both the oracle/sun and google java coding guidelines specify. It is also (I believe) the default behavior when using the the intellij java formatter.